client/app/components/Parameters.jsx

```diff
@@ -111,8 +111,9 @@ export class Parameters extends React.Component {
   applyChanges = () => {
     const { onValuesChange, disableUrlUpdate } = this.props;
     this.setState(({ parameters }) => {
+      const parametersWithPendingValues = parameters.filter(p => p.hasPendingValue);
       forEach(parameters, p => p.applyPendingValue());
```

Comment:
Now we can use `parametersWithPendingValues` here too :)

Reply:
Not sure I understand. Are you referring to using `parametersWithPendingValues` in the following line? As in
```js
forEach(parametersWithPendingValues, p => p.applyPendingValue());
```

If that's what you mean, I deliberately avoided that in order to future-proof someone removing the `hasPendingValue` check in `applyPendingValue` without knowing of this external concern.

Reply:
I wouldn't be that concerned about it (doesn't seem a likely thing or something that would cause a big issue here). But not a big deal in any case :)